src/lib.rs

```diff
@@ -261,171 +285,92 @@ fn validate(paths: [&str; 3]) -> Result<ValidationResults, Box<dyn Error>> {
-                if let Some(i) = check_script(&dent)? {
-                    results.invalid_scripts.push(i);
-                }
+            // TODO Script checks don't really fit nicely into InvalidItem, refactor needed to log
+            // more details about the invalid script's ABI.
+            if num_public_script_methods > 1 {
+                results.invalid_items.push(InvalidItem {
+                    kind: Validator::Script,
+                    file: dent.path().display().to_string(),
+                    text: String::new(),
+                });
             }
         }
     }
     Ok(results)
 }
 
-fn check_test(
-    matcher: &RegexMatcher,
-    dent: &walkdir::DirEntry,
-    lnum: u64,
-    line: &str,
-) -> Option<InvalidItem> {
-    // We are guaranteed to find a match, so the unwrap is ok.
-    let the_match = matcher.find(line.as_bytes()).unwrap().unwrap();
-    let text = line[the_match].to_string();
-
-    // Check if it matches our pattern.
-    let pattern = r"test(Fork)?(Fuzz)?_(Revert(If_|When_){1})?\w+\(";
-    let validator = RegexMatcher::new_line_matcher(pattern).unwrap();
-
-    // If match is found, test name is good, otherwise it's bad.
-    let match_result = validator.find(text.as_bytes()).unwrap();
-    if match_result.is_some() {
-        return None
+fn is_valid_test_name(name: &str) -> bool {
+    if !name.starts_with("test") {
+        return true // Not a test function, so return.
     }
-
-    let trimmed_test = text.trim();
-    let item = InvalidItem {
-        kind: Validator::Test,
-        file: dent.path().to_str().unwrap().to_string(),
-        line: lnum,
-        // Trim off the leading "function " and remove the trailing "(".
-        text: trimmed_test[9..trimmed_test.len() - 1].to_string(),
-    };
-
-    Some(item)
-}
-
-fn check_src_fn(
-    matcher: &RegexMatcher,
-    dent: &walkdir::DirEntry,
-    lnum: u64,
-    line: &str,
-) -> Option<InvalidItem> {
-    // We are guaranteed to find a match, so the unwrap is ok.
-    let the_match = matcher.find(line.as_bytes()).unwrap().unwrap();
-    let text = line[the_match].to_string();
-
-    // Ensure public/external functions have no leading underscore, and internal/private functions
-    // have a leading underscore.
-    let vis_validator = RegexMatcher::new_line_matcher(r"\b(public|external)\b").unwrap();
-    let is_public = vis_validator.find(text.as_bytes()).unwrap();
-    let name = text[9..text.len() - 1].to_string();
-    let first_char = name.trim().chars().next()?;
-
-    if is_public.is_some() && first_char != '_' || is_public.is_none() && first_char == '_' {
-        return None
-    }
-
-    let item = InvalidItem {
-        kind: Validator::Src,
-        file: dent.path().to_str().unwrap().to_string(),
-        line: lnum,
-        // Trim off the leading "function " and remove the trailing "(".
-        text: name,
-    };
-    Some(item)
+    let regex = Regex::new(r"test(Fork)?(Fuzz)?_(Revert(If_|When_){1})?\w+").unwrap();
```

Comment:
can make this lazy static

Reply:
per https://blessed.rs guidance, went with `once_cell` over `lazy_static` and use their recommended macro in f1e17f7